src/pybind/mgr/smb/handler.py

```diff
@@ -886,7 +886,10 @@ def _check_share(
                     msg=f"earmark has already been set by {earmark.split('.')[0]}",
                 )
             # Check if earmark is set by same cluster
-            if earmark.split('.')[2] != share.cluster_id:
+            if (
+                earmark != EarmarkTopScope.SMB.value
+                and earmark.split('.')[2] != share.cluster_id
+            ):
```

Comment:
This doesn't look right to me. If the top level scope is not equal to smb then the previous if statement should be true and an exception should be raised. 

I think you are hitting the case where the existing earmark is actually just `smb`. There's no additional scopes to split on.  I hate to keep harping on the same stuff but if you parse the earmark exactly once using a single function and return a structured object you can do comparsons in a clearer manner rather than ad-hoc (repeat) parsing of the earmark string.